third_party/move/move-compiler-v2/tests/checking/typing/struct_no_field_list.move

```diff
@@ -0,0 +1,15 @@
+module 0x42::m {
+
+    struct S has copy, drop;
+
+    fun f(_s: S): S {
+        // dropping paramter, constructing new value
+        S
+    }
+    
+    fun d() {
+        let S = S;
```

Comment:
Do we really want to enable these?  They do nothing, so are likely to be some sort of user error if they appear.

Reply:
I'm not sure what you mean with enable. This possible from the logic of the language, whether it makes sense to write something like this is another question. Rust certainly allows this.